src/redux/actions.js

```diff
@@ -15,3 +15,27 @@ export const saveActiveItem = (id) => {
     payload: { id: id },
   };
 };
+
+export const fetchServerListWithRouting =
+  (navigate, cookies) => async (dispatch) => {
+    const token = cookies.get('authToken', { path: '/' });
+    if (!token) {
+      navigate('login');
+      console.log('no token');
+    } else {
+      try {
+        const response = await fetch('http://localhost:80/servers', {
+          method: 'GET',
+          headers: {
+            Authorization: `Bearer ${token}`,
+            'Content-Type': 'application/json',
+          },
+        });
+        const data = await response.json();
+        const servers = data.data;
+        dispatch(creationServers(servers));
```

Comment:
this action is called to add servers to the list. In this case we are getting the full list of servers, so need to use another action here to replace the entire list with new data